pkl-gradle/src/main/java/org/pkl/gradle/task/EvalTask.java

```diff
@@ -39,27 +48,44 @@ public abstract class EvalTask extends ModulesTask {
   @Optional
   public abstract Property<String> getModuleOutputSeparator();
 
-  @OutputDirectory
-  @Optional
-  public abstract DirectoryProperty getMultipleFileOutputDir();
-
   @Input
   @Optional
   public abstract Property<String> getExpression();
 
+  @Inject
+  public abstract ObjectFactory getObjectFactory();
+
+  private CliEvaluator getCliEvaluator() {
+    return new CliEvaluator(
```

Comment:
Note that this results in multiple `CliEvaluator` instances, each of which recomputes `CliEvaluator.outputFiles` and `CliEvaluator.outputDirectories`.